### PR TITLE
Prevent unnecessary re-renders of checkbox component

### PR DESCRIPTION
**Pull request by petrican**

### CODE REVIEW

**Comment by petrican** [comment]
Screenshot 2020-01-16 at 16 12 28
Prevents all those 560 re-renders to just 2

File: src/components/widget/Checkbox.js [hunk 1]
```diff
@@ -42,9 +53,9 @@ const Checkbox = props => {
         <input
           ref={rawWidget}
           type="checkbox"
-          checked={widgetData[0].value}
+          checked={checkedState}
           disabled={widgetData[0].readonly || disabled}
-          onChange={e => handlePatch(widgetField, e.target.checked, id)}
+          onChange={e => updateCheckedState(e)}
```
Comment (siemiatj) [inline src/components/widget/Checkbox.js]: why do we need an anonymous function here ?

**Comment by siemiatj** [comment]
Just one thing. Other than that, LGTM

File: src/components/widget/Checkbox.js (hunk 1 above)
Comment (petrican) [reply src/components/widget/Checkbox.js]: This function is used to do fine tuning to state before further calling the handlePatch. In our case set the state value checkedState L:34 .

Comment (siemiatj) [reply src/components/widget/Checkbox.js]: I still don't understand why there's an anonymous function there. If param set in onChange is just e and updateCheckedState expects e as well, why not just onChange={updateCheckedState} ?

**Pull request merged**